compiler/noirc_frontend/src/parser/parser/types.rs

```diff
@@ -181,9 +181,9 @@ pub(super) fn int_type() -> impl NoirParser<UnresolvedType> {
 pub(super) fn named_type<'a>(
     type_parser: impl NoirParser<UnresolvedType> + 'a,
 ) -> impl NoirParser<UnresolvedType> + 'a {
-    path_no_turbofish().then(generic_type_args(type_parser)).map_with_span(|(path, args), span| {
-        UnresolvedTypeData::Named(path, args, false).with_span(span)
-    })
+    path(type_parser.clone()).then(generic_type_args(type_parser)).map_with_span(
+        |(path, args), span| UnresolvedTypeData::Named(path, args, false).with_span(span),
+    )
```

Comment:
This would allow `Foo::<Bar><Baz>`, right? We should probably error in that case

Reply:
Good point! It turns out `path_no_turbofish()` here was fine. For some reason I thought this is where struct patterns are parsed, but that happens in a different place.